lib/browser/guest-view-manager.js

```diff
@@ -340,35 +340,35 @@ const isWebViewTagEnabled = function (contents) {
   return isWebViewTagEnabledCache.get(contents)
 }
 
-const handleMessage = function (channel, handler) {
-  ipcMainInternal.on(channel, (event, ...args) => {
-    if (isWebViewTagEnabled(event.sender)) {
-      handler(event, ...args)
-    } else {
-      event.returnValue = null
-    }
-  })
+const assertWebViewTagEnabled = function (contents) {
+  if (!isWebViewTagEnabled(contents)) {
+    throw new Error('<webview> disabled')
+  }
 }
 
-handleMessage('ELECTRON_GUEST_VIEW_MANAGER_CREATE_GUEST', function (event, params, requestId) {
-  event._replyInternal(`ELECTRON_RESPONSE_${requestId}`, createGuest(event.sender, params))
+ipcMainUtils.handle('ELECTRON_GUEST_VIEW_MANAGER_CREATE_GUEST', function (event, params) {
+  assertWebViewTagEnabled(event.sender)
+  return createGuest(event.sender, params)
 })
 
-handleMessage('ELECTRON_GUEST_VIEW_MANAGER_CREATE_GUEST_SYNC', function (event, params) {
-  event.returnValue = createGuest(event.sender, params)
+ipcMainUtils.handle('ELECTRON_GUEST_VIEW_MANAGER_CREATE_GUEST_SYNC', function (event, params) {
+  assertWebViewTagEnabled(event.sender)
+  return createGuest(event.sender, params)
 })
 
-handleMessage('ELECTRON_GUEST_VIEW_MANAGER_DESTROY_GUEST', function (event, guestInstanceId) {
+ipcMainInternal.on('ELECTRON_GUEST_VIEW_MANAGER_DESTROY_GUEST', function (event, guestInstanceId) {
```

Comment:
maybe a silly question, why are some of these `ipcMainUtils.handle` and others are `ipcMainInternal.on`?

Reply:
I've restored the `handleMessage` helper which uses `ipcMainInternal.handle` consistently